core/src/avm2/globals/flash/display/display_object.rs

```diff
@@ -21,7 +20,43 @@ use crate::vminterface::Instantiator;
 use crate::{avm2_stub_getter, avm2_stub_setter};
 use ruffle_render::filters::Filter;
 use std::str::FromStr;
-use swf::BlendMode;
+use swf::Twips;
+use swf::{BlendMode, Rectangle};
+
+pub fn display_object_allocator<'gc>(
+    class: ClassObject<'gc>,
+    _activation: &mut Activation<'_, 'gc>,
+) -> Result<Object<'gc>, Error<'gc>> {
+    // todo we want an Err here
+    // as a script can do `new DisplayObject()`
+    panic!("This allocator should not be called! {:?}", class);
```

Comment:
Error here: `ArgumentError: Error #2012: DisplayObject$ class cannot be instantiated.`

Reply:
This should never be hit - the actual error gets thrown from `DisplayObject.as`

Reply:
This can still be hit by a class that extends DisplayObject, which is unintentional. See this SWF for example, which panics with this PR but runs fine (no errors) on master, logging `[object DO]`:
[avm2-tests.swf.zip](https://github.com/ruffle-rs/ruffle/files/11521182/avm2-tests.swf.zip)

Both this PR and master's state is incorrect, though- in FP, it errors with `ArgumentError: Error #2012: DO$ class cannot be instantiated.`

Reply:
Good catch - I've added a fix and a test case.